qiskit_experiments/base_experiment.py

```diff
@@ -110,7 +110,7 @@ def run(
         else:
             # Validate experiment is compatible with existing data container
             metadata = experiment_data.metadata()
-            if metadata.get("experiment_data") != self._type:
+            if metadata.get("experiment_type") != self._type:
```

Comment:
Not sure if it's correct. Probably not, according to @merav-aharoni in #124. Need to check.

Reply:
Has this been resolved?

Reply:
According to @chriseclectic in #124 this is the fix